### PR TITLE
fix(Datagrid): expand listener added

### PR DESCRIPTION
**Pull request by jagathgj**
Contributes to #2603 

Event listener function added to expandable rows.

#### What did you change?
ExpandableRow.stories.js
Datagrid/index.js
useRowExpandListener.js

#### How did you test and verify your work?
storybook

### CODE REVIEW

**Review by davidmenendez — APPROVED** [review]
code wise this looks good to me. not sure if anyone who's more familiar with datagrid wants to chime in though.

File: packages/ibm-products/src/components/Datagrid/useRowExpandListener.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+const useRowExpandListener = (hooks) => {
+  const getProps = (props, { row }) => {
+    return {
+      ...props,
+      onClick: () => {
+        if (row.original.onRowExpand) {
+          row.original.onRowExpand(row);
+        }
```
Comment (AlexanderMelox) [inline packages/ibm-products/src/components/Datagrid/useRowExpandListener.js:17]:
```suggestion
          row?.original?.onRowExpand?.(row);
```

I would make this less bloated

**Comment by AlexanderMelox** [comment]
@jagathgj I think it should be fine to just add it to `useExpandedRow`, the onExpandRow can live in there. I don't see a need to make another hook just to listen to the row click 🤔 Any thoughts on this @matthewgallo

**Review by matthewgallo — CHANGES_REQUESTED** [review]
Agree with @AlexanderMelox, I don't think this necessarily is something that needs it's own hook. There are other extensions which add a listener without the need for a completely new hook.

**Comment by elycheea** [comment]
@jagathgj Sorry for the delay on this one — seems like we keep missing it, but looks like we have a conflict with one of the recent PRs now but once we resolve, @matthewgallo can help re-review this.

**Review by matthewgallo — CHANGES_REQUESTED** [review]
To follow the pattern we've set with other event handlers like this, I think it makes most sense to have `onRowExpand` be passed directly to the `useDatagrid` hook and it returns the row that was expanded. As opposed to this approach which adds it directly on the row within the data.

**Review by matthewgallo — APPROVED** [review]

**Pull request merged**